src/main/java/software/amazon/cloudwatchlogs/emf/sinks/TCPClient.java

```diff
@@ -31,12 +31,13 @@
     private boolean shouldConnect = true;
 
     public TCPClient(Endpoint endpoint) {
-        socket = createSocket();
         this.endpoint = endpoint;
     }
 
     private void connect() {
         try {
+            // Avoid "socket already connected" error (https://issues.amazon.com/issues/P54323886)
```

Comment:
Remove the link from this comment. Also, the comment might not even be needed. If we keep it, you should be more descriptive as to why we want to create a new socket on every connect() call.